src/app/core/entity/entity.ts

```diff
@@ -112,6 +112,28 @@ export class Entity {
     this._id = Entity.createPrefixedId(this.getType(), newEntityId);
   }
 
+  /**
+   * Returns an array of strings by which the entity can be searched.
+   *
+   * By default the parts of the "name" property (split at spaces) is used if it is present.
+   *
+   * <b>Overwrite this method in subtypes if you want an entity type to be searchable by other properties.</b>
+   */
+  @DatabaseField() get searchIndices(): string[] {
+    let indices = [];
+
+    // default indices generated from "name" property
+    if (this.hasOwnProperty("name")) {
+      indices = this["name"].split(" ");
+    }
+
+    return indices;
+  }
+  set searchIndices(value) {
```

Comment:
Is there a reason why this function is implemented?

Reply:
SchemaService is writing all fields' values and I received errors when I didn't define a setter. Alternatively we could introduce a special case in the schemaService for searchIndices again and not load them - but I preferred to keep it general there and add this peculiar empty setter here.

Reply:
Ok makes sense